src/kms/entry.rs

```diff
@@ -14,9 +43,11 @@ pub struct KeyParams {
 
     /// An optional external reference for the key
     #[serde(default, rename = "ref", skip_serializing_if = "Option::is_none")]
-    pub reference: Option<String>,
+    pub reference: Option<KeyReference>,
```

Comment:
I expected that the reference field would be something like "secure_element:{key_id}" and the data field would be empty in this case, but I suppose this works too and might be better for indexing keys.

Reply:
I am not too set on either approach. I can leave it as-is or change it to w/e is your preferred method.